WISE-Community/WISE-Client
# build(Client): Build using GitHub Actions

## Changes

- Changed ```get-testing-status.sh``` to show the status of all our testing pipelines since you can only really use one at a time
  - private-wise-api-pipeline
  - private-wise-client-github-actions-pipeline
  - private-wise-client-pipeline

- Updated ```start-testing.sh``` to use the new pipeline that uses the GitHub Actions builds
- Updated ```start-testing.sh``` to check if any testing pipeline is currently in use and prevent starting the pipeline

- Changed ```stop-testing.sh``` to stop the new pipeline

## Test

I have set up another PR to use for testing this PR. In the other PR, you can see the Build check if you click "Show all checks". If you click on Details for the Build, you can see the Build Id which looks something like this
```wise-client-github-actions-project:6da08232-9077-4fa1-89d1-f06fb0aab492```
- https://github.com/WISE-Community/WISE-Client/pull/776

The branch name for the other PR is ```github-actions-code-build-test-2```

- Run ```scripts/get-testing-status.sh``` and make sure it shows the status of all our testing pipelines
- Run the start testing script with the --dry-run argument to see the commit hash and build id that would be used if we were to actually start the testing pipeline
  - ```scripts/start-testing.sh github-actions-code-build-test-2 --dry-run```
- Run the start testing script for real to start the testing pipeline. It should take about 8 minutes or less before you can start testing. We are saving about 5 minutes since the builds are pre-built now.
  - ```scripts/start-testing.sh github-actions-code-build-test-2```
- Try running the start testing script again while the pipeline is currently being used. It should not let you start the pipeline again.
  - ```scripts/start-testing.sh github-actions-code-build-test-2```
- Once the servers are ready for testing, as a student, go to an Open Response step. You should see the text "Hello World" above the Save button to verify that the server is actually using the build from ```github-actions-code-build-test-2```
- Run the stop testing script to stop the servers
  - ```scripts/stop-testing.sh```
